File: back-tp-final/src/main/java/org/example/backtpfinal/entities/Report.java
package org.example.backtpfinal.entities;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;
import java.util.UUID;

@Entity
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Report {
    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private UUID id;
    private int workedHoursRate;
    private LocalDateTime startDate;
    private LocalDateTime endDate;

    @ManyToOne
    @JoinColumn(name="employee_id")
    private Employee employee;


    public void setId(UUID id) {
        this.id = id;
    }

    public UUID getId() {
        return id;
    }
}

File: admin-app-tp-final/src/models/Report.js
class Report {
    constructor(id,workedHours,startDate,endDate,employeeId) {
        this.id = id
        this.workedHours = workedHours
        this.startDate = startDate
        this.endDate = endDate
        this.employeeId = employeeId
    }


    get id() {
        return this.id
    }
    get workedHours() {
        return this.workedHours
    }
    get startDate() {
        return this.startDate
    }
    get endDate() {
        return this.endDate
    }
    get employeeId() {
        return this.employeeId
    }

}
export default Report